src/Aspire.Hosting.Azure/AzurePublishingContext.cs

```diff
@@ -310,6 +313,27 @@ void CaptureBicepOutputsFromParameters(IResourceWithParameters resource)
         {
             if (resource.GetDeploymentTargetAnnotation() is { } annotation && annotation.DeploymentTarget is AzureBicepResource br)
             {
+                // Materialize Dockerfile factory if present
+                if (resource.TryGetLastAnnotation<DockerfileBuildAnnotation>(out var dockerfileBuildAnnotation) &&
```

Comment:
I'm feeling some type of way about the fact that any component that interacts with the Dockerfile has to know to invoke this path. Maybe we can encapsulate all this into a public API that deployers/publishers can call?

Reply:
The logic isn't always the same, but we can share some code. I'll do it in a follow up.